### PR TITLE
Ports: Add the Small Device C Compiler (WIP)

### PR DESCRIPTION
**Pull request by kleinesfilmroellchen**
After having the pleasure of talking to the SDCC maintainer at FrOSCon, I decided to port their cool niche compiler to Serenity!

This current version fails to configure because of unknown configure script weirdness. Feel free to test and send improvements my way.

### CODE REVIEW

File: Ports/sdcc/package.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=sdcc
+version=20220820-13663
+useconfigure=true
```
Comment (timschumi) [inline Ports/sdcc/package.sh:4]:
```suggestion
port='sdcc'
version='20220820-13663'
useconfigure='true'
```

File: Ports/sdcc/patches/0001-Recognize-Serenity-as-a-system.patch [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+diff --git a/config.sub b/config.sub
+index 1d8e98b..6ed7f57 100755
+--- a/config.sub
++++ b/config.sub
+@@ -1390,6 +1390,7 @@ case $os in
+ 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
+ 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
+ 	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
++			| -serenity* \
```
Comment (ADKaster) [inline Ports/sdcc/patches/0001-Recognize-Serenity-as-a-system.patch:18]: This should be the use fresh config sub option in package.sh, not a patch

File: Ports/sdcc/patches/0001-Recognize-Serenity-as-a-system.patch [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ 	      | -midnightbsd*)
+ 	# Remember, each alternative MUST END IN *, to match a version number.
+ 		;;
```
Comment (timschumi) [inline Ports/sdcc/patches/0001-Recognize-Serenity-as-a-system.patch:21]: This should probably be replaced with `use_fresh_config_sub`/`config_sub_paths`.

File: Ports/sdcc/patches/0001-Recognize-Serenity-as-a-system.patch (hunk 2 above)
Comment (kleinesfilmroellchen) [reply Ports/sdcc/patches/0001-Recognize-Serenity-as-a-system.patch:18]: You need a host though, it doesn't configure without one either

**Review by timschumi — CHANGES_REQUESTED** [review]
Haven't looked into the actual configure failure yet, just some generic Ports stuff that I noticed.

File: Ports/sdcc/patches/0001-Recognize-Serenity-as-a-system.patch (hunk 3 above)
Comment (kleinesfilmroellchen) [reply Ports/sdcc/patches/0001-Recognize-Serenity-as-a-system.patch:21]: Again; the config script needs a host argument or it doesn't work!

Comment (timschumi) [reply Ports/sdcc/patches/0001-Recognize-Serenity-as-a-system.patch:21]: Yes? That's exactly what `use_fresh_config_sub` does. It downloads a copy of `config.sub` from the GNU git server (which has Serenity support already included) to replace the outdated copy that is shipped inside the tarball, no manual patching needed.

Comment (kleinesfilmroellchen) [reply Ports/sdcc/patches/0001-Recognize-Serenity-as-a-system.patch:21]: ah okay, is that even documented? I couldn't find it in the port documentation.

Comment (timschumi) [reply Ports/sdcc/patches/0001-Recognize-Serenity-as-a-system.patch:21]: [Yes](https://github.com/SerenityOS/serenity/blob/master/Ports/README.md#use_fresh_config_sub), although it doesn't explicitly mention that this is a viable replacement for patching in Serenity support manually.

**Pull request closed without merging**